### PR TITLE
Platform that combines all behaviors

### PR DESCRIPTION
**Pull request by danielbe1**
The demo scene is Demo.scene and the platform is to the right of the players.

- There are some quirks with the platform when the player rolls under it as  a ball it will treat the platform as a solid even when it is semi solid.
- Dropping from the platform doesn't seem to work right because the groundcheck doesn't detect the platform for some reason, the only way to get the character to drop is to press both crouch and space and pray.

### CODE REVIEW

File: Assets/Scripts/Map/Platform/Dynamic/DynamicPlatformConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+using System;
+using NaughtyAttributes;
```
Comment (Dejin1) [inline Assets/Scripts/Map/Platform/Dynamic/DynamicPlatformConfiguration.cs]: One thing I have seen when using Naughty attributes is when you go to build the project it causes errors to finish the build I had to remove the NaughtyAttribute.DLL and the actors folder for it to be succesful I can screenshot this but do you perhaps know why?

File: Assets/Scripts/Map/Platform/Dynamic/Components/DynamicPlatformParenting.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        InitTopEdgeCollider();
+    }
+
+    private void OnDestroy()
+    {
+        if (_topEdgecollider != null)
+            Destroy(_topEdgecollider);
+    }
+    
+    private void OnTriggerEnter2D(Collider2D other)
```
Comment (Dejin1) [inline Assets/Scripts/Map/Platform/Dynamic/Components/DynamicPlatformParenting.cs]: Can this be integrated with the Trigger script?

File: Assets/Scripts/Map/Platform/Dynamic/Components/DynamicPlatformParenting.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        
+        other.transform.SetParent(transform);
+    }
+
+    private void OnTriggerStay2D(Collider2D other)
+    {
+        OnTriggerEnter2D(other);
+    }
+
+    private void OnTriggerExit2D(Collider2D other)
```
Comment (Dejin1) [inline Assets/Scripts/Map/Platform/Dynamic/Components/DynamicPlatformParenting.cs]: same question as line 43 about the integration into the trigger script

File: Assets/Scripts/Map/Platform/Dynamic/RectangularDynamicPlatform.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    private Rigidbody2D _rigidbody2D;
+
+    private List<Vector2> _topFacePoints;
+    
+    public override void Initialize()
+    {
+        var bounds = _mainBodyCollider.bounds;
+        var width = bounds.size.x / transform.localScale.x;
+        var height = bounds.size.y / transform.localScale.y;
+        var topLeft = new Vector3(-width / 2, height / 2);
```
Comment (Dejin1) [inline Assets/Scripts/Map/Platform/Dynamic/RectangularDynamicPlatform.cs]: Is there ever a situation where width or height would not be divided in half?

File: Assets/Scripts/Player/CharacterController2D.cs [hunk 5]
```diff
@@ -39,8 +39,8 @@ public class CharacterController2D : MonoBehaviour
 
     public List<Collider2D> BodyColliders = new List<Collider2D>();             // Colliders that make up the characters body and are used to represent its volume in the physics space.
 
-    private List<SemiSolidPlatform> m_semiSolidPatformsStoodOn =                // All semi-solid platforms the character is standing on and are considered "ground".
-        new List<SemiSolidPlatform>();                
+    private List<Platform> m_semiSolidPatformsStoodOn =                         // All semi-solid platforms the character is standing on and are considered "ground".
```
Comment (Dejin1) [inline Assets/Scripts/Player/CharacterController2D.cs]: Why the change of name and addition of space just curious?

File: Assets/Scripts/Map/Platform/Dynamic/DynamicPlatformConfiguration.cs (hunk 1 above)
Comment (danielbe1) [reply Assets/Scripts/Map/Platform/Dynamic/DynamicPlatformConfiguration.cs]: Really? I didn't notice it when I used to build the web player version which should be way more strict about these things. Ill check it out.

File: Assets/Scripts/Map/Platform/Dynamic/Components/DynamicPlatformParenting.cs (hunk 2 above)
Comment (danielbe1) [reply Assets/Scripts/Map/Platform/Dynamic/Components/DynamicPlatformParenting.cs]: How do you mean? It looks like the trigger script does way more than what the platform needs to operate right and it can be configured to break the platform behavior.

File: Assets/Scripts/Map/Platform/Dynamic/RectangularDynamicPlatform.cs (hunk 4 above)
Comment (danielbe1) [reply Assets/Scripts/Map/Platform/Dynamic/RectangularDynamicPlatform.cs]: Not for this kind of platform, its how the bounds class defines its corners.

File: Assets/Scripts/Player/CharacterController2D.cs (hunk 5 above)
Comment (danielbe1) [reply Assets/Scripts/Player/CharacterController2D.cs]: Nothing intended, ill remove it

File: Assets/Scripts/Player/JinPlayerMovement.cs [hunk 6]
```diff
@@ -85,7 +85,7 @@ void Start()
     void Update()
     {
         shootProjectile = inputStringForProjectile;
-        getDownFromPlatform = crouch && Input.GetButtonDown(stringGetDownFromPlatform);
+        getDownFromPlatform = crouch && Input.GetButton(stringGetDownFromPlatform);
 
         horizontalDirection = GetAxis(leftKey, rightKey);
```
Comment (Dejin1) [inline Assets/Scripts/Player/JinPlayerMovement.cs]: This is an old script if you look at develops latest build of this script you will see its not correct

File: Assets/Scripts/Map/Platform/Dynamic/Components/DynamicPlatformParenting.cs (hunk 2 above)
Comment (Dejin1) [reply Assets/Scripts/Map/Platform/Dynamic/Components/DynamicPlatformParenting.cs]: Yeah Its just a suggestion not needed ill resolve this

File: Assets/Scripts/Map/Platform/Dynamic/RectangularDynamicPlatform.cs (hunk 4 above)
Comment (Dejin1) [reply Assets/Scripts/Map/Platform/Dynamic/RectangularDynamicPlatform.cs]: Alright just checking im resolving this

File: Assets/Scripts/Player/CharacterController2D.cs (hunk 5 above)
Comment (Dejin1) [reply Assets/Scripts/Player/CharacterController2D.cs]: Ok ill resolve this

File: Assets/Scripts/Map/Platform/Dynamic/DynamicPlatformConfiguration.cs (hunk 1 above)
Comment (Dejin1) [reply Assets/Scripts/Map/Platform/Dynamic/DynamicPlatformConfiguration.cs]: Ok let me know if your able to replicate the issue if not ill send you a gif
